fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/Migration.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * (C) Copyright IBM Corp. 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.ibm.fhir.database.utils.model;
+
+import java.util.List;
+
+import com.ibm.fhir.database.utils.api.IDatabaseStatement;
+
+/**
+ * A database migration step
+ */
+public interface Migration {
+    public List<IDatabaseStatement> migrateFrom(Integer priorVersion);
```

Comment:
I think documenting the method. 

Starting from the prior version, the Migration implements as a set of steps that are sequentially executed. 

Also the set of IDatabaseStatements, maybe they should indicate if each step is idempotent. 

Also a minor comment, to be consistent in this code base, shouldn't it be IMigration.

Reply:
added this javadoc:
```
    /**
     * Migrate from some previous version to the current version of this database object
     *
     * @param priorVersion
     *            the version being migrated from
     * @return a list of statements to be executed sequentially to bring a database object from the prior version to the
     *         current version
     * @implSpec this method should only be executed when the current version > priorVersion and so the steps need not be
     *         idempotent
     */
```

> Also a minor comment, to be consistent in this code base, shouldn't it be IMigration.
I'm pretending I didn't see this part ;-). @JohnTimm